includes/options/class-amp-options-menu.php

```diff
@@ -423,6 +436,191 @@ private function list_template_conditional_options( $options, $parent = null ) {
+		unset( $errors_by_sources['plugin']['gutenberg'] ); // Omit Gutenberg to prevent unintentional attribution for shortcodes.
+		unset( $errors_by_sources['plugin']['amp'] ); // Omit AMP because disabling in AMP responses would be bad!
+		if ( isset( $errors_by_sources['plugin'] ) ) {
+			return $errors_by_sources['plugin'];
+		}
+		return [];
+	}
+
+	/**
+	 * Get suppressible plugin sources.
+	 *
+	 * @return string[] Plugin sources which are suppressible.
+	 */
+	private static function get_suppressible_plugin_sources() {
+		return array_unique(
+			array_intersect(
+				array_merge(
+					array_keys( self::get_plugin_errors_by_sources() ),
+					array_keys( AMP_Options_Manager::get_option( Option::SUPPRESSED_PLUGINS ) )
+				),
+				array_map(
+					function ( $plugin_file ) {
+						return strtok( $plugin_file, '/' );
+					},
+					get_option( 'active_plugins', [] )
+				)
+			)
+		);
+	}
+
+	/**
+	 * Render suppressed plugins.
+	 *
+	 * @since 1.6
+	 */
+	public function render_suppressed_plugins() {
+		$suppressed_plugins = AMP_Options_Manager::get_option( Option::SUPPRESSED_PLUGINS );
+
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		$plugins = get_plugins();
+		unset( $plugins['amp/amp.php'] );
+		foreach ( array_keys( $plugins ) as $plugin_file ) {
+			if ( ! is_plugin_active( $plugin_file ) ) {
+				unset( $plugins[ $plugin_file ] );
+			}
+		}
+		uasort(
+			$plugins,
+			static function ( $a, $b ) {
+				return strcmp( $a['Name'], $b['Name'] );
+			}
+		);
+
+		?>
+		<fieldset>
+			<h4 class="title hidden"><?php esc_html_e( 'Suppressed Plugins', 'amp' ); ?></h4>
+			<p>
+				<?php esc_html_e( 'When a plugin emits invalid markup that causes an AMP validation error, one option is to review the invalid markup and allow it to be removed. Another option is to suppress the plugin from doing anything when rendering AMP pages. What follows is the list of active plugins with any causing validation errors being highlighted. If a plugin is emitting invalid markup that is causing validation errors and this plugin is not necessary on the AMP version of the page, it can be suppressed.', 'amp' ); ?>
+			</p>
+
+			<style>
+			.amp-suppressed-plugins .suppressed-plugin:checked + label {
+				text-decoration: line-through;
+			}
+			.amp-suppressed-plugins .plugin > details {
+				margin-left: 30px;
+			}
+			.amp-suppressed-plugins .plugin > details > ul {
+				margin-left: 30px;
+				margin-top: 0.5em;
+				margin-bottom: 1em;
+				list-style-type: disc;
+			}
+			.amp-suppressed-plugins summary {
+				user-select: none;
+				cursor: pointer;
+			}
+			</style>
+
+			<?php
+			$element_name = AMP_Options_Manager::OPTION_NAME . '[' . Option::SUPPRESSED_PLUGINS . ']';
+
+			$errors_by_sources = self::get_plugin_errors_by_sources()
+			?>
+			<ul>
+				<?php foreach ( $plugins as $plugin_file => $plugin ) : ?>
+					<?php
+					$plugin_slug = strtok( $plugin_file, '/' );
```

Comment:
What about in (rare) cases where multiple plugins can be found in the same directory?

Reply:
Yeah, it's a good point. The reality is that `AMP_Validation_Manager::get_source()` is not able to detect such nested plugins anyway. The source slug extracted from a plugin will always be just the directory name if the plugin is not a single file:

https://github.com/ampproject/amp-wp/blob/9d662d07da228c6cc8dedfab5abc724feaf39177/includes/validation/class-amp-validation-manager.php#L1636-L1639

So that is why the the directory alone is being considered (or else just the file name if the plugin is not in a directory).